### ISSUE TITLE
Is possible to add some background highlight for code blocks

### ISSUE THREAD

**Issue by tonyriverms** [Lukasa/language-restructuredtext#65]
Is it possible to add some background highlight to a code block like this?

<img width="780" alt="capture" src="https://user-images.githubusercontent.com/9701455/40581540-78dc4032-6129-11e8-879f-fcab3a7dbaa5.PNG">

**Comment by Alhadis** [comment]
I'd like to say _“you can use your [stylesheet](https://flight-manual.atom.io/using-atom/sections/basic-customization/#style-tweaks)”_, but it turns out that adding a background colour to certain lines in Atom is more complicated than simply dropping this into your stylesheet:

~~~less
.line > * > .syntax--text.syntax--restructuredtext{
	&:only-child > .syntax--source.syntax--embedded.syntax--python{
		background-color: #236423;
	}
}
~~~

Unless you're happy with the effect looking like this:

<img src="https://user-images.githubusercontent.com/2346707/182376669-a1741c64-8b39-44b5-b2df-f1887df76dbc.png" alt="Figure 1" width="549" />

If not, the effect you're after is achieved with something a lot like this:

~~~js
const ed = atom.workspace.getActiveTextEditor();
const range = ed.getLastSelection().getBufferRange();
const marker = ed.markBufferRange(range);
const decoration = ed.decorateMarker(marker, {
	type: "line",
	class: "python-snippet",
})
~~~

… and this code needs to be run every time an update is made to an editor that's using `text.restructuredtext` as its grammar. And the code to do that monitoring is really, really, _really_ fiddly. But if you're masochistic and/or determined enough to do down that route, the relevant APIs have something resembling documentation [here](https://flight-manual.atom.io/api/v1.57.0/TextEditor/#instance-decorateMarker).

tl;dr: Atom's APIs suck ass.
